src/main/java/com/github/gilday/darkmode/DarkModeSync.java

```diff
@@ -26,16 +29,27 @@
  */
 public final class DarkModeSync implements Disposable {
 
-  private final ScheduledFuture<?> scheduledFuture;
+  private ScheduledFuture<?> scheduledFuture;
 
-  private final LafManager lafManager;
-  private final DarkModeSyncThemes themes;
-  private final Alarm updateOnUIThreadAlarm;
+  private LafManager lafManager;
+  private DarkModeSyncThemes themes;
+  private Alarm updateOnUIThreadAlarm;
 
-  /** @param lafManager IDEA look-and-feel manager for getting and setting the current theme */
-  public DarkModeSync(final LafManager lafManager) {
+  private static final class MyStartupActivity implements StartupActivity.DumbAware {
+    @Override
+    public void runActivity(@NotNull Project project) {
+      getInstance(project).onStartup();
+    }
+  }
+
+  public static DarkModeSync getInstance(@NotNull Project project) {
+    return project.getService(DarkModeSync.class);
```

Comment:
This creates a new `DarkModeSync` service for each Project, right? We want one `DarkModeSync` singleton at the application level

Reply:
You are right, this will end up creating a service instance for each project. Good catch!